File: example/Phenology_Model/C#/Phase.cs
using System;
public static class Phase
{
	public static String getPhaseAsString(double p)
	{		
		if (p < 0)
		{
			return "BeforeSowing";
		}
		else if (p >= 0 && p < 1)
		{
			return "SowingToEmergence";
		}
		else if (p >= 1 && p < 2)
		{
			return "EmergenceToFloralInitiation";
		}
		else if (p >= 2 && p < 3)
		{
			return "FloralInitiationToHeading";
		}
		else if (p >= 3 && p < 4)
		{
			return "HeadingToAnthesis";
		}
		else if (p >= 4 && p <4.5)
		{
			return "AnthesisToEndCellDivision";
		}
		else if (p >= 4.5 && p < 5)
		{
			return "EndCellDivisionToEndGrainFill";
		}
		else if (p >= 5 && p < 6)
		{
			return "EndGrainFillToMaturity";
		}
		else if (p >= 6 && p < 7)
		{
			return "AllOver";
		}
		else
		{
			return "NoPhase";
		}
	}

	public static GrowthStage PreviousMoment(double phaseValue)
	{
		if (phaseValue < 0)
		{
			return GrowthStage.Unknown;
		}
		else if (phaseValue >= 0 && phaseValue < 1)//SowingToEmergence
		{
			return GrowthStage.ZC_00_Sowing;
		}
		else if (phaseValue >= 1 && phaseValue < 2)//EmergenceToFloralInitiation
		{
			return GrowthStage.ZC_10_Emergence;
		}
		else if (phaseValue >= 2 && phaseValue < 3)//FloralInitiationToAnthesis
		{
			return GrowthStage.FloralInitiation;
		}
		else if (phaseValue >= 3 && phaseValue < 4)//FloralInitiationToAnthesis
		{
			return GrowthStage.Heading;
		}
		else if (phaseValue == 4)//AnthesisToEndCellDivision
		{
			return GrowthStage.ZC_65_Anthesis;
		}
		else if (phaseValue == 4.5)//EndCellDivisionToEndGrainFill
		{
			return GrowthStage.ZC_75_EndCellDivision;
		}
		else if (phaseValue >= 5 && phaseValue < 6)//EndGrainFillToMaturity
		{
			return GrowthStage.ZC_91_EndGrainFilling;
		}
		else if (phaseValue >= 6 && phaseValue < 7)//AllOver
		{
			return GrowthStage.ZC_92_Maturity;
		}
		else
		{
			throw new InvalidOperationException();
		}

	}

}

File: old_example/Phenology_Model/Python/Phase.py
'''
Created on 17 sept. 2018

@author: midingoy
'''

def getPhaseAsString(p):
    if (p < 0): return "BeforeSowing";
    elif (p >= 0 and p < 1): return "SowingToEmergence";
    elif (p >= 1 and p < 2):  return "EmergenceToFloralInitiation";
    elif (p >= 2 and p < 3): return "FloralInitiationToHeading";
    elif (p >= 3 and p < 4):  return "HeadingToAnthesis";
    elif  (p >= 4 and p <4.5) :return "AnthesisToEndCellDivision";
    elif  (p >= 4.5 and p < 5):return "EndCellDivisionToEndGrainFill";
    elif  (p >= 5 and p < 6):return "EndGrainFillToMaturity";
    elif  (p >= 6 and p < 7):return "AllOver";
    else:return "NoPhase";

def PreviousMoment(phaseValue):
    if (phaseValue < 0):return "Unknown";
    elif (phaseValue >= 0 and phaseValue < 1): return "ZC_00_Sowing";
    elif (phaseValue >= 1 and phaseValue < 2):return "ZC_10_Emergence";
    elif (phaseValue >= 2 and phaseValue < 3):return "FloralInitiation";
    elif (phaseValue >= 3 and phaseValue < 4):return "Heading";
    elif (phaseValue == 4):return "ZC_65_Anthesis";
    elif (phaseValue == 4.5):return "ZC_75_EndCellDivision";
    elif (phaseValue >= 5 and phaseValue < 6):return "ZC_91_EndGrainFilling";
    elif (phaseValue >= 6 and phaseValue < 7):return "ZC_92_Maturity";



if __name__ == '__main__':
    pass